src/cashlib/cashlib.cpp

```diff
@@ -477,3 +501,460 @@ SLAPI unsigned int SmGetError(void *smId)
     ScriptMachineData *smd = (ScriptMachineData *)smId;
     return (unsigned int)smd->sm->getError();
 }
+
+// result must be 32 bytes
+extern "C" void sha256(const unsigned char *data, unsigned char len, unsigned char *result)
+{
+    CSHA256 sha;
+    sha.Write(data, len);
+    sha.Finalize(result);
+}
+
+
+// result must be 32 bytes
+extern "C" void hash256(const unsigned char *data, unsigned char len, unsigned char *result)
+{
+    CHash256 hash;
+    hash.Write(data, len);
+    hash.Finalize(result);
+}
+
```

Comment:
could be wrong but `hash256` and `sha256` seems almost identical to me. Do we need both?

Reply:
hash256 is a double sha256.  I don't want to get too deeply into what we "need" and into the wisdom of the hash256 name choice which is causing this confusion.  I wanted to preserve the terminology used in bitcoin code and dev documentation and move on.